core/src/main/java/org/apache/cxf/io/DelayedCachedOutputStreamCleaner.java

```diff
@@ -74,25 +78,146 @@ default void forceClean() {
+                }
+            }
+            if (oldTimerTask != null) {
+                oldTimerTask.cancel();
+            }
+            clean(closeables);
+        }
+
+        @Override
+        public void forceClean() {
+            TimerTask oldTimerTask = null;
+            synchronized (lock) {
+                clean(queue);
+                if (timerTask != null) {
+                    oldTimerTask = timerTask;
+                    timerTask = null;
+                }
+            }
+            if (oldTimerTask != null) {
+                oldTimerTask.cancel();
+            }
+        }
+
+        @Override
+        public void close()  {
+            TimerTask oldTimerTask = null;
+            synchronized (lock) {
+                queue.clear();
+                if (timerTask != null) {
+                    oldTimerTask = timerTask;
+                    timerTask = null;
+                }
+            }
+            if (oldTimerTask != null) {
+                oldTimerTask.cancel();
+            }
+        }
+
+        @Override
+        public int size() {
+            return queue.size();
+        }
+
+        private void clean(Collection<DelayedCloseable> closeables) {
+            final Iterator<DelayedCloseable> iterator = closeables.iterator();
+            while (iterator.hasNext()) {
+                final DelayedCloseable next = iterator.next();
+                try {
+                    iterator.remove();
+                    LOG.warning("Unclosed (leaked?) stream detected: " + next.closeable.hashCode());
+                    next.closeable.close();
+                } catch (final IOException | RuntimeException ex) {
+                    LOG.warning("Unable to close (leaked?) stream: " + ex.getMessage());
+                }
+            }
+        }
+    }
+
+    private static final class DefaultDelayedCleaner implements DelayedCleaner {
         private final long delay; /* default is 30 minutes, in milliseconds */
         private final DelayQueue<DelayedCloseable> queue = new DelayQueue<>();
         private final Timer timer;
-        
-        DelayedCleanerImpl(final long delay) {
+        private final AtomicBoolean scheduled = new AtomicBoolean(false);
+
+        DefaultDelayedCleaner(final long delay) {
             this.delay = delay;
             this.timer = new Timer("DelayedCachedOutputStreamCleaner", true);
-            this.timer.scheduleAtFixedRate(new TimerTask() {
-                @Override
-                public void run() {
-                    clean();
-                }
-            }, 0, Math.max(MIN_DELAY, delay >> 1));
+            
         }
 
         @Override
         public void register(Closeable closeable) {
             queue.put(new DelayedCloseable(closeable, delay));
+            // Initialize timer lazily only when at least one closeable is registered
+            if (scheduled.compareAndSet(false, true)) {
+                this.timer.scheduleAtFixedRate(new TimerTask() {
```

Comment:
So the Timer should be constructed here instead for the timer thread to be lazily started on the first registered closeable.  And note the DefaultDelayedCleaner.close call would then need a timer != null check.